src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.example.onlinebookstore.repository.impl;
+
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
+    public void save(Book book) {
+        if (book.getId() == null) {

```

Comment:
let's control transactions
remove book.getId() == null because you shouldn't provide id here
let's catch DataProcessingException if something happen

Reply:
same for other methods